ci/test-clang.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+set -eu
+
+trap 'rm -fr "$toolchain"' EXIT
+
+# Set up workspace
+rootdir=${GITHUB_WORKSPACE:-$(dirname "$(dirname "$(realpath "$0")")")}
+toolchain=$rootdir/toolchain
+
+# Pull toolchain out of container
+echo "[+] Downloading toolchain from container"
+docker create --name llvm-project ghcr.io/clangbuiltlinux/llvm-project:stage2
+mkdir "$toolchain"
+docker cp llvm-project:/usr/local/bin "$toolchain"
+docker cp llvm-project:/usr/local/include "$toolchain"
+docker cp llvm-project:/usr/local/lib "$toolchain"
+echo "[+] Cleaning up container"
+docker rm llvm-project
+
+# Test toolchain in Docker images so that the environment is consistent,
+# regardless of runners.
+docker_images=(
+    docker.io/fedora:latest
+)
+# Arch Linux is x86_64-only
```

Comment:
Why? Also, we don't have any other arch builds yet.

Reply:
> Why?

That might be better worded at "Arch Linux is an x86_64 only distribution"?

> Also, we don't have any other arch builds yet.

Key word is yet... I'd like to stop adding x86_64-isms everywhere but I'll just punt that to later down the road.

Reply:
I didn't know that about arch. A little surprised TBH.

Reply:
But the proposed updated wording is clearer.

Reply:
> I didn't know that about arch. A little surprised TBH.

There is technically Arch Linux for ARM (https://archlinuxarm.org) and i686 (https://archlinux32.org/) but they are ports, rather than officially supported, so I don't tend to count them much.

> But the proposed updated wording is clearer.

Would you rather I drop the `uname -m` check right now or just update the wording? I am fine with whatever.

Reply:
Keep the `uname -m` check, but update the wording. (Without the `uname -m` check, this would need to be added back when adding aarch64 host support, right?)

Reply:
> Keep the `uname -m` check, but update the wording.

Done: 162cc51a36827e94e2d538a35916171a156c871f

> Without the `uname -m` check, this would need to be added back when adding aarch64 host support, right?)

Correct!